src/wp-includes/block-template-utils.php

```diff
@@ -1432,39 +1432,48 @@ function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '
 	$template_hierarchy[] = 'index';
 	return $template_hierarchy;
 }
+
 /**
  * Inject ignoredHookedBlocks metadata attributes into a template or template part.
  *
- * Given a `wp_template` or `wp_template_part` post object, locate all blocks that have
+ * Given an object that represents a `wp_template` or `wp_template_part` post object
+ * prepared for inserting or updating the database, locate all blocks that have
  * hooked blocks, and inject a `metadata.ignoredHookedBlocks` attribute into the anchor
  * blocks to reflect the latter.
  *
- * @param WP_Post $post A post object with post type set to `wp_template` or `wp_template_part`.
- * @return WP_Post The updated post object.
+ * @since 6.5.0
+ * @access private
+ *
+ * @param stdClass        $post    An object representing a template or template part
+ *                                 prepared for inserting or updating the database.
+ * @param WP_REST_Request $request Request object.
+ * @return stdClass The updated object representing a template or template part.
  */
-function inject_ignored_hooked_blocks_metadata_attributes( $post ) {
+function inject_ignored_hooked_blocks_metadata_attributes( $post, $request ) {
+	$filter_name = current_filter();
+	if ( ! str_starts_with( $filter_name, 'rest_pre_insert_' ) ) {
+		return $post;
+	}
+	$post_type = str_replace( 'rest_pre_insert_', '', $filter_name );
```

Comment:
@ockham what do you think to the below? Its a solution which doesn't rely on us hard-coding part of the filter name (`rest_pre_insert_`). I feel like it makes it less brittle.

```suggestion
	// Match the text after 'wp_' in the filters name.
	$pattern = '/wp_(.*)$/';
	preg_match( $pattern, $filter_name, $result );
	$post_type = isset( $result[0] ) ? $result[0] : '';
	
	if ( '' === $post_type ) {
		return $post;
	}
```

Reply:
Different philosophies I guess 🤔 My approach was to be as restrictive as possible; the filter would bail early if ever hooked to a different hook. (If a developer were to use a different hook deliberately, we'd have to update the check accordingly; there's a slight potential for confusion there if a developer tries to hook it to a different hook without checking the function body and gets puzzled when it's not invoked. Is this what makes it feel brittle to you? My thinking was that that wasn't a big risk, since people would be quick to check the function body.)

Generally, I prefer things to break visibly rather than subtly, which is why I chose this approach; you hook to a different hook, your filter isn't run at all. (It's admittedly a bit debatable here if "not running a filter" is that visible.)

I find matching the `wp_` a bit risky, since it might "work" for different filters that happen to include that infix. OTOH, it wouldn't cover any post types that _don't_ include `wp_`. Right now, this isn't relevant. In the future, we might at least expand the reach of this filter to also cover the `wp_navigation` case (rather than doing that from a specialized function in the Navigation block code), as [has been requested](https://wordpress.slack.com/archives/C02RQBWTW/p1709650462126059?thread_ts=1709649022.232249&cid=C02RQBWTW). _Even_ further down the road, we might extend Block Hooks to work with content, i.e. posts, pages, etc; those block types wouldn't have the `wp_` prefix.

Right now, all of this is very academic, of course 😬

Reply:
Hmm yeah after seeing it from your POV I do have to agree with your thinking here. Lets leave it as-is for now. Thank you!